### DISCUSSION TITLE
CURL_OPENSSL_4 and CURL_OPENSSL_3

### DISCUSSION THREAD

**Discussion by vikrambhagwat** [curl/curl#11112] [Q&A] [answered]
Hello folks,
 I tried below command  
  
   nm -D libcurl.so.4.6.0 |grep CURL*

On Linux system: CURL_OPENSSL_3 
On other Linux system: CURL_OPENSSL_4 

Any idea what it means? 

-- Vikram

**Comment by vikrambhagwat** [comment]
I spent some time with curl versioning. Seems like

CURL_OPENSSL_3 = libcurl version 3 built with openssl
CURL_OPENSSL_4 = libcurl version 4 built with openssl

Please correct me here in case...

Now how to map libcurl version 3 and libcurl version 4 with 
curl versions which are 7.xx or 8.xxx?

**Reply by bagder** [reply, marked as answer]
> CURL_OPENSSL_3 = libcurl version 3 built with openssl
> CURL_OPENSSL_4 = libcurl version 4 built with openssl

They are symbols provided by *some builds* of libcurl, and they are for the **SONAME** 3 and 4.

> Now how to map libcurl version 3 and libcurl version 4 with
> curl versions which are 7.xx or 8.xxx?

That is only possible by checking how those builds are made and when they decided to bump their SONAME from 3 to 4.

I guess you're looking at Debian/Ubuntu. They decided by themselves that they should not follow the curl project's own way of deciding the SONAME numbering so you cannot get help by curl's git history for this.
